test/module/irohad/torii/torii_queries_test.cpp

```diff
@@ -98,8 +102,260 @@ TEST_F(ToriiServiceTest, FindWhereQueryServiceSync) {
   ASSERT_EQ(response.error_response().reason(), "Not valid");
 }
 
+/**
+ * Tests for account response
+ */
+
+TEST_F(ToriiServiceTest, FindAccountWhenStatefulInvalid) {
+  EXPECT_CALL(svMock, validate(A<const iroha::model::Query &>()))
+      .WillRepeatedly(Return(true));
+
+  iroha::model::Account account;
+  account.account_id = "accountB";
+
+  EXPECT_CALL(wsv_query, getAccount(_)).WillRepeatedly(Return(account));
+
+  iroha::protocol::QueryResponse response;
+
+  auto query = iroha::protocol::Query();
+  query.set_creator_account_id("accountA");
+  query.mutable_get_account()->set_account_id("accountB");
+
+  auto stat = torii_utils::QuerySyncClient(Ip, Port).Find(query, response);
+  ASSERT_TRUE(stat.ok());
+  // Should not return Error Response because stateless valid
+  ASSERT_NE(response.error_response().reason(), "Not valid");
+  // Must be invalid due to failed stateful validation caused by no permission
+  // to read account
+  ASSERT_EQ(response.error_response().reason(), "Not valid query");
+}
+
+TEST_F(ToriiServiceTest, FindAccountWhenValid) {
+  EXPECT_CALL(svMock, validate(A<const iroha::model::Query &>()))
+      .WillRepeatedly(Return(true));
+
+  iroha::model::Account account;
+  account.account_id = "accountA";
+
+  EXPECT_CALL(wsv_query, getAccount(_)).WillRepeatedly(Return(account));
+
+  iroha::protocol::QueryResponse response;
+
+  auto query = iroha::protocol::Query();
+  query.set_creator_account_id("accountA");
+  query.mutable_get_account()->set_account_id("accountA");
```

Comment:
Can user `accountA` send valid query which has `creator_account_id = "accountB"` and `GetAccount.account_id = "accountB"`?